### DISCUSSION TITLE
Stat Analysis NBRCTC Aggregation Issue

### DISCUSSION THREAD

**Discussion by lindsayrblank** [dtcenter/METplus#1826] [Statistical Computation] [answered]
Hello all,

I'm having a strange problem with Stat_Analysis. I am trying to aggregate the `NBRCTC` line type. I am running MET version 10.0.0 Docker as a Singularity image on an HPC. When I run the following command, I get the following error:

```
singularity exec /path/to/singularity_image/met.sif stat_analysis -job aggregate -line_type NBRCTC -lookin /path/to/data/directory -out_stat ./test.stat -by FCST_VAR,FCST_THRESH,FCST_LEAD,FCST_INIT_HOUR -fcst_init_hour 00  -set_hdr DESC 'NA' -rank_corr_flag FALSE -n_boot_rep 0 -out_bin_size 1 -v 3

 Creating STAT output file "./test.stat"
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_S_construct null not valid
Aborted
```

When I run the same command with `-line_type CTC`, it works perfectly. There are both `CTC` and `NBRCTC` lines in the input files.

Any help would be greatly appreciated!

Best,
Lindsay

**Comment by JohnHalleyGotway** [comment, marked as answer]
@lindsayrblank thanks for letting us know about this runtime error. @georgemccabe found a pretty obvious bug in the code that is the source of this issue starting on this [line of code](https://github.com/dtcenter/MET/blob/c37e95c62e41b0c89020117b484158280e31b1fc/met/src/libcode/vx_analysis_util/stat_job.cc#L2020).
```
         case stat_nbrctc: write_header_row       (nbrctc_columns, n_sl1l2_columns, 1,      stat_at, 0, 0); break;
         case stat_nbrcts: write_header_row       (nbrcts_columns, n_sl1l2_columns, 1,      stat_at, 0, 0); break;
         case stat_nbrcnt: write_header_row       (nbrcnt_columns, n_sl1l2_columns, 1,      stat_at, 0, 0); break;
```
It's a copy/paste error and the fix is easy.

FYI, I'm able to replicate this error by running:
```
MET-main_v10.0/met/bin/stat_analysis -lookin grid_stat_APCP_12_120000L_20050807_120000V.stat -job aggregate -line_type NBRCTC -out_stat out_stat.txt
```
Produces this:
```
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_M_construct null not valid
Abort trap: 6
```
But if I change `-out_stat` to just `-out`, the error goes away:
```
MET-main_v10.0/met/bin/stat_analysis -lookin grid_stat_APCP_12_120000L_20050807_120000V.stat -job aggregate -line_type NBRCTC -out out.txt
```
So this is obviously a bug that we need to fix (see dtcenter/MET#2271), but I just wanted to point out a temporary workaround.

I assume you'll want a bugfix for MET version 10.0, or would providing on for MET version 10.1 be sufficient?

Thanks,
John

**Reply by lindsayrblank** [reply]
Thank you, @JohnHalleyGotway. I tested your work around and it worked on my instance. A MET version 10.0 bugfix would be preferable to 10.1, if you're able to.
